front/src/config/store.js

```diff
@@ -165,6 +172,18 @@ const fetchSubmissionUntilCorrectionEnds = (submissionId, timeout = 60) => {
   };
 };
 
+const changeMyProfile = (profile) => {
+  return {type: CHANGE_MY_PROFILE, profile };
+}
+
+const fetchMyProfile = () => {
+  return async (dispatch) => {
+    await userApi.getMyProfile()
+      .catch(() => dispatch(changeMyProfile(false)))
+      .then(profile => dispatch(changeMyProfile(profile)));
+  }
+};
```

Comment:
`await` allows you not to use .then and .catch methods of a promise.
Please save the response as a `const` and catch the potential errors with a `try{}except{}` statement ;)

Reply:
made a `try{...} catch {...}` block